### PR TITLE
Implement the Secret game preset.

### PR DESCRIPTION
**Pull request by moonheart08**
## About the PR
This is a prerequisite for many gamemodes, like nukies and revolutionaries.

:cl:
- add: Added the Secret game preset, which randomly selects the game rules to play.

### CODE REVIEW

File: Content.Server/GameTicking/Rules/SecretRuleSystem.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    [Dependency] private readonly GameTicker _ticker = default!;
+
+    public override string Prototype => "Secret";
+
+    private readonly IReadOnlyDictionary<string, float> _ruleTable = new Dictionary<string, float>
+    {
+        { "Traitor", 0.75f },
+    };
+
+    private readonly List<string> _attachedRules = new();
```
Comment (mirrorcult) [inline Content.Server/GameTicking/Rules/SecretRuleSystem.cs:22]: does this need to be a list? looks like it only selects one rule

File: Resources/Prototypes/game_presets.yml [hunk 2]
```diff
@@ -7,6 +7,16 @@
   showInVote: true
   description: extended-description
 
+- type: gamePreset
+  id: Secret
+  alias:
+    - secret
```
Comment (mirrorcult) [inline Resources/Prototypes/game_presets.yml:13]: sekrit

**Review by mirrorcult — APPROVED** [review]

File: Content.Server/GameTicking/Rules/SecretRuleSystem.cs (hunk 1 above)
Comment (moonheart08) [reply Content.Server/GameTicking/Rules/SecretRuleSystem.cs:22]: instinctual future proofing moment
I'll make it a singular string.

File: Content.Server/GameTicking/Rules/SecretRuleSystem.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly GameTicker _ticker = default!;
+
+    public override string Prototype => "Secret";
+
+    private readonly IReadOnlyDictionary<string, float> _ruleTable = new Dictionary<string, float>
+    {
+        { "Traitor", 0.75f },
+    };
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/SecretRuleSystem.cs:20]: Looks like this only gets converted to a list anyway so may as well just make it a list of tuples?

File: Content.Server/GameTicking/Rules/SecretRuleSystem.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            if (!_random.Prob(rule.Value))
+                continue;
+
+            if (_ticker.AddGameRule(_prototypeManager.Index<GameRulePrototype>(rule.Key)))
+            {
+                _attachedRule = rule.Key;
+                break;
+            }
+        }
+    }
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/SecretRuleSystem.cs:50]: image

**Comment by ElectroJr** [comment]
Will this be superseeded by #6550, or is this still useful?

**Comment by mirrorcult** [comment]
conflicts

**Comment by moonheart08** [comment]
Prolly not going to be fixing this, so

**Pull request closed without merging**